### ISSUE TITLE
Get "WARNING: the subvolume is read-write and has received_uuid set" on ro snapshots

### ISSUE THREAD

**Issue by CorrosiveTruths** [kdave/btrfs-progs#419]
Hi, is the logic for getting the read-write with received_uuid warning the wrong way around and detecting read-only with received_uuid?

```
# btrfs sub create snap
Create subvolume './snap'
# btrfs sub snap -r snap snap_ro
Create a readonly snapshot of 'snap' in './snap_ro'
# btrfs send snap_ro | btrfs rec snap
At subvol snap_ro
At subvol snap_ro
# btrfs sub show snap/snap_ro
WARNING: the subvolume is read-write and has received_uuid set,
         don't use it for incremental send. Please see section
         'SUBVOLUME FLAGS' in manual page btrfs-subvolume for
         further information.
root/root/btrfs-progs/snap/snap_ro
        Name:                   snap_ro
        UUID:                   fe844a7b-b395-c841-98b5-7e0bdd61b8d8
        Parent UUID:            -
        Received UUID:          cce9ac06-e20b-7446-8316-fc60869a8822
        Creation time:          2021-11-01 20:26:03 +0000
        Subvolume ID:           6927
        Generation:             130498
        Gen at creation:        130497
        Parent ID:              6925
        Top level ID:           6925
        Flags:                  **readonly**
        Send transid:           130495
        Send time:              2021-11-01 20:26:03 +0000
        Receive transid:        130498
        Receive time:           2021-11-01 20:26:03 +0000
        Snapshot(s):

```

**Comment by kdave** [comment]
Out of all the rw/received combinations, the read-only+received still reports the warning, I've reproduced it here. So yeah it's wrong and was not covered by the test case.

**Comment by kdave** [comment]
The logic is correct but there are different origins of the read-only flag checked. There's 
BTRFS_ROOT_SUBVOL_RDONLY a raw root item flag with value 0x1 and there's another one obtained by ioctl, BTRFS_SUBVOL_RDONLY, and it's 0x2.

**Comment by kdave** [comment]
Thanks for the report, now fixed in devel and queued for the 5.15 release.
